sdk/formrecognizer/azure-ai-formrecognizer/migration-guide.md

```diff
@@ -0,0 +1,255 @@
+# Guide for migrating to `azure-ai-formrecognizer (4.0.0-beta.1)` from `azure-ai-formrecognizer (3.0.x)`
+
+This guide is intended to assist in the migration to `azure-ai-formrecognizer (4.0.0-beta.1)` from `azure-ai-formrecognizer (3.0.x)`. It will focus on side-by-side comparisons for similar operations between the two packages.
+
+We assume that you are familiar with the old SDK `azure-ai-formrecognizer (3.0.x)`. If not, please refer to the new SDK README for [azure-ai-formrecognizer][README] directly rather than this migration guide.
+
+## Table of contents
+
+- [Guide for migrating to `azure-ai-formrecognizer (4.0.0-beta.1)` from `azure-ai-formrecognizer (3.0.x)`](#guide-for-migrating-to-azure-ai-formrecognizer-400-beta1-from-azure-ai-formrecognizer-30x)
+    - [Table of contents](#table-of-contents)
+    - [Migration benefits](#migration-benefits)
+        - [Cross Service SDK improvements](#cross-service-sdk-improvements)
+    - [Important changes](#important-changes)
+        - [Instantiating clients](#instantiating-clients)
+        - [Analyze Documents using a Prebuilt Model](#analyze-documents-using-a-prebuilt-model)
+        - [Build a custom document analysis model](#build-a-custom-document-analysis-model)
+        - [Analyze Documents using a Custom Model](#analyze-documents-using-a-custom-model)
+        - [Manage models](#manage-models)
+    - [Additional samples](#additional-samples)
+
+## Migration benefits
+
+A natural question to ask when considering whether to adopt a new version or library is its benefits. As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the Java
+client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. The most important is that many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To improve the development experience across Azure services, a set of uniform [design guidelines][Guidelines] was created for all languages to drive a
+consistent experience with established API patterns for all services. A set of [Java design guidelines][GuidelinesJava] was introduced to ensure that Java clients have a natural and idiomatic feel with respect to the Java ecosystem. Further details are available in the guidelines
+for those interested.
```

Comment:
might just be me but this paragraph feels really general / seems like it's explaining the track 1 to track 2 migration. We changed our design to adopt the new service improvements, but I guess you could extrapolate to say that some of it was due to developer feedback.